cactusbot/handlers/events.py

```diff
@@ -102,29 +106,74 @@ def __init__(self, cache_data, api):
     async def on_host(self, packet):
         """Handle host packets."""
 
-        if self.alert_messages["host"]["announce"]:
-            return MessagePacket(
-                self.alert_messages["host"]["message"].replace(
-                    "%USER%", packet.user
-                ))
+        if not self.alert_messages["host"]["announce"]:
+            return
+
+        response = MessagePacket(
+            self.alert_messages["host"]["message"].replace(
+                "%USER%", packet.user
+            ))
+
+        if packet.success:
```

Comment:
Arguments! :smile: 

```python
def check_cache(cache, packet):
    # things

if self.cache_hosts:
    if check_cache(self.cached_hosts, packet):
        return response
return response
```

You could also move `self.cached_THINGS` into `self.cache`.

```python
self.cache = {
    "follows": {},
    "hosts": {},
}
```